src/main/webapp/app/shared/markdown-editor/monaco/markdown-editor-monaco.component.ts

```diff
@@ -0,0 +1,362 @@
+import { faGripLines, faQuestionCircle } from '@fortawesome/free-solid-svg-icons';
+import { v4 as uuid } from 'uuid';
+import { FileUploaderService } from 'app/shared/http/file-uploader.service';
+import { AlertService, AlertType } from 'app/core/util/alert.service';
+import { MonacoEditorActionGroup } from 'app/shared/monaco-editor/model/actions/monaco-editor-action-group.model';
+import { MonacoHeadingAction } from 'app/shared/monaco-editor/model/actions/monaco-heading.action';
+import { MonacoFullscreenAction } from 'app/shared/monaco-editor/model/actions/monaco-fullscreen.action';
+import { MonacoColorAction } from 'app/shared/monaco-editor/model/actions/monaco-color.action';
+import { ColorSelectorComponent } from 'app/shared/color-selector/color-selector.component';
+import { CdkDragMove, Point } from '@angular/cdk/drag-drop';
+import { MonacoEditorDomainAction } from 'app/shared/monaco-editor/model/actions/monaco-editor-domain-action.model';
+import { MonacoEditorDomainActionWithOptions } from 'app/shared/monaco-editor/model/actions/monaco-editor-domain-action-with-options.model';
+
+interface MarkdownActionsByGroup {
+    standard: MonacoEditorAction[];
+    header: MonacoHeadingAction[];
+    color?: MonacoColorAction;
+    domain: {
+        withoutOptions: MonacoEditorDomainAction[];
+        withOptions: MonacoEditorDomainActionWithOptions[];
+    };
+    meta: MonacoEditorAction[];
+}
+
+// TODO: Once the old markdown editor is gone, remove the style url.
+@Component({
+    selector: 'jhi-markdown-editor-monaco',
+    templateUrl: './markdown-editor-monaco.component.html',
+    styleUrls: ['./markdown-editor-monaco.component.scss', '../markdown-editor.component.scss'],
+})
+export class MarkdownEditorMonacoComponent implements AfterContentInit, AfterViewInit, OnDestroy {
+    @ViewChild(MonacoEditorComponent, { static: false }) monacoEditor: MonacoEditorComponent;
+    @ViewChild('wrapper', { static: true }) wrapper: ElementRef<HTMLDivElement>;
+    @ViewChild('fileUploadFooter', { static: false }) fileUploadFooter?: ElementRef<HTMLDivElement>;
+    @ViewChild('resizablePlaceholder', { static: false }) resizePlaceholder?: ElementRef<HTMLDivElement>;
+    @ViewChild('resizeHandle', { static: false }) resizeHandle?: ElementRef<HTMLDivElement>;
+    @ViewChild('actionPalette', { static: false }) actionPalette?: ElementRef<HTMLElement>;
+    @ViewChild(ColorSelectorComponent, { static: false }) colorSelector: ColorSelectorComponent;
+
+    @Input()
+    set markdown(value: string | undefined) {
+        this._markdown = value;
+        this.monacoEditor?.setText(value ?? '');
+    }
+
+    _markdown?: string;
+
+    @Input()
+    enableFileUpload = true;
+
+    @Input()
+    enableResize = true;
+
+    /**
+     * The initial height the editor should have. If set to 'external', the editor will try to grow to the available space.
+     */
+    @Input({ required: true })
+    initialEditorHeight: MarkdownEditorHeight | 'external';
```

Comment:
one could use just have the input not required and if it is undefined the editor grows to the available space

Reply:
I did this on purpose to prevent situations where the developer does not provide a value here and ends up with a (potentially infinitely) growing editor.

What do you think?

Reply:
I see - that makes sense